### PR TITLE
[4.0] Add message to menu

### PR DESCRIPTION
**Pull request by brianteeman**
Due to the restructured menu in J4 there was no direct link on the menu or the components dashboard to the list off aall Private Messages

This PR fixes that

### CODE REVIEW

**Comment by infograf768** [comment]
Thanks. I had noted that issue in https://github.com/joomla/joomla-cms/issues/26067#issuecomment-526552331

I have a problem in PhpMyadmin to apply the Insert from the update sql:
It says
`1064 - You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ' 17, 20, 0, '*', 1, '0000-00-00 00:00:00', '0000-00-00 00:00:00')' at line 1`

The error comes from the fact that the `params` column is missing an empty value by using `''`
`'class:messages', 0, , 17, 20, 0,`
should be
`'class:messages', 0, '', 17, 20, 0,`

same for postgres update

Once corrected, _and I don't know if it is my version of PHPMyadmin or not_, I still have an error:
 ` #1292 - Incorrect datetime value: '0000-00-00 00:00:00' for column 'checked_out_time' at row 1`

**Comment by alikon** [comment]
for the datetime issue for postgresql the defualt date time is not `0000-00-00 00:00:00'` but `1970-01-01 00:00:00`  for mysql phpmyadmin depending on your mysql version and setting the correct datetime should be `'1000-01-01 00:00:00'` see https://dev.mysql.com/doc/refman/8.0/en/datetime.html for more info

File: installation/sql/postgresql/joomla.sql [hunk 1]
```diff
@@ -1344,6 +1344,8 @@ SELECT 16, 'main', 'com_tags', 'Tags', '', 'Tags', 'index.php?option=com_tags&vi
...
 SELECT 17, 'main', 'com_associations', 'Multilingual Associations', '', 'Multilingual Associations', 'index.php?option=com_associations&view=associations', 'component', 1, 1, 1, "extension_id", 0, '1970-01-01 00:00:00', 0, 0, 'class:associations', 0, '', 31, 32, 0, '*', 1 FROM "#__extensions" WHERE "name" = 'com_associations';
 INSERT INTO "#__menu" ("id", "menutype", "title", "alias", "note", "path", "link", "type", "published", "parent_id", "level", "component_id", "checked_out", "checked_out_time", "browserNav", "access", "img", "template_style_id", "params", "lft", "rgt", "home", "language", "client_id")
+SELECT 18, 'main', 'com_messages_manage', 'Private Messages', '', 'Messaging/Private Messages', 'index.php?option=com_messages&view=messages', 'component', 1, 10, 2, `extension_id`, 0, '0000-00-00 00:00:00', 0, 0, 'class:messages-add', 0, '', 18, 19, 0, '*', 1 FROM `#__extensions` WHERE `name` = 'com_messages';
```
Comment (alikon) [inline installation/sql/postgresql/joomla.sql]:
```suggestion
SELECT 18, 'main', 'com_messages_manage', 'Private Messages', '', 'Messaging/Private Messages', 'index.php?option=com_messages&view=messages', 'component', 1, 10, 2, `extension_id`, 0, '1970-01-01 00:00:00', 0, 0, 'class:messages-add', 0, '', 18, 19, 0, '*', 1 FROM `#__extensions` WHERE `name` = 'com_messages';
```

**Comment by infograf768** [comment]
@alikon
You are damned right. Issue with update is solved here using `'1000-01-01 00:00:00'`
My sql version is `Server version: 5.7.25 - MySQL Community `

Shall not we correct this all over our sqls? Including joomla.sql ?

**Comment by alikon** [comment]
No we don't need in joomla for now at least for mysql < 8, cause we are setting the no sql strict mode https://github.com/joomla/joomla-cms/blob/fe597ef10fe6d6f0841bebc8aeebaad72d7080f3/libraries/joomla/database/driver/mysqli.php#L177

**Comment by brianteeman** [comment]
should be ok now. it was late when I did this and it sadly was pretty sloppy

**Comment by alikon** [comment]
my only concern is about the lft,rgt  values as the `#__menu` is one of the nested tables,
but take my comment like as someone reading only the code in hurry , (ie without proof) :smile

**Comment by brianteeman** [comment]
i was concerned too. didnt seem to have a negative impact and i dont know how to fix it without changing all records

**Comment by brianteeman** [comment]
The system test errors reported by drone seem valid to me but are unrelated to this PR

**Comment by infograf768** [comment]
Drone relaunched

**Comment by wilsonge** [comment]
Thanks!

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**